app/views/developers/index.html.erb

```diff
@@ -0,0 +1,37 @@
+<h2>List developers</h2>
+<%= form_for  @developerearchform, url: developers_path, method: :get, html: { class: "form-inline"}  do |f| %>
```

Comment:
When I search with language: jp, I see the following logs.
This is super inefficient from the database point of view.
If there're a million records, the query hits the database a milliton times.
Sometimes this kind of query consumes all the resources of the database and make the database stop.
As a server side developer, you definitely have to avoid this.
Can you fix this?
Can you also share it with the other developers in Da Nang for them not to make the same mistake?
Also can you share this with other developers? 
```
$ tail -f log/development.log
:
Started GET "/developers?utf8=%E2%9C%93&developer_search_form%5Blanguage_id%5D=1&developer_search_form%5Bprogramming_language_id%5D=&commit=Search" for 127.0.0.1 at 2018-04-03 11:37:29 +0900
Processing by DevelopersController#index as HTML
  Parameters: {"utf8"=>"✓", "developer_search_form"=>{"language_id"=>"1", "programming_language_id"=>""}, "commit"=>"Search"}
  Rendering developers/index.html.erb within layouts/application
   (0.2ms)  SELECT "languages"."code", "languages"."id" FROM "languages"
   (0.3ms)  SELECT "programming_languages"."name", "programming_languages"."id" FROM "programming_languages"
  SQL (0.5ms)  SELECT "developers"."id" AS t0_r0, "developers"."email" AS t0_r1, "developers"."created_at" AS t0_r2, "developers"."updated_at" AS t0_r3, "developer_languages"."id" AS t1_r0, "developer_languages"."developer_id" AS t1_r1, "developer_languages"."language_id" AS t1_r2, "developer_languages"."created_at" AS t1_r3, "developer_languages"."updated_at" AS t1_r4 FROM "developers" INNER JOIN "developer_languages" ON "developer_languages"."developer_id" = "developers"."id" WHERE "developer_languages"."language_id" = $1 ORDER BY "developers"."created_at" DESC  [["language_id", 1]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 101]]
   (0.7ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 101]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 98]]
   (0.4ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 98]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 92]]
   (0.4ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 92]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 89]]
   (0.4ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 89]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 86]]
   (0.3ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 86]]
   (0.3ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 85]]
   (0.5ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = "developer_programming_languages"."programming_language_id" WHERE "developer_programming_languages"."developer_id" = $1  [["developer_id", 85]]
   (0.4ms)  SELECT "languages"."code" FROM "languages" INNER JOIN "developer_languages" ON "languages"."id" = "developer_languages"."language_id" WHERE "developer_languages"."developer_id" = $1  [["developer_id", 82]]
   (0.3ms)  SELECT "programming_languages"."name" FROM "programming_languages" INNER JOIN "developer_programming_languages" ON "programming_languages"."id" = 
```

Reply:
@haiphantony 
I wonder if this is fixed.
Maybe not.
Can you check the log?